src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -133,6 +137,10 @@ public IndexContext(TableMetadata tableMeta, IndexMetadata config)
 
         this.primaryKeyFactory = Version.LATEST.onDiskFormat().primaryKeyFactory(tableMeta.comparator);
 
+        this.segmentCompactionEnabled = Boolean.parseBoolean(
+            config.options.getOrDefault(ENABLE_SEGMENT_COMPACTION_OPTION_NAME, "true")
```

Comment:
Would it be safer to have the default as `false`? The option is called `enable_segment_compaction` so I would expect that if the option was not applied then it would be disabled.

Reply:
But then it wouldn't be backwards compatible. We also all agree that disabling segment compaction is generally a bad idea in most cases. 

I can change the option name to `disable_segment_compaction`? However I don't recall any other `disable_` options, so that would look a bit weird.

Reply:
Ok, I'm happy leave this as is. We can always change the default later when we have completed performance testing of the change. Also, when / if we complete the async index compaction this option should go completely.